test/node-sync-test.js

```diff
@@ -0,0 +1,164 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('bsert');
+const FullNode = require('../lib/node/fullnode');
+const WalletNode = require('../lib/wallet/node');
+const {forValue} = require('./util/common');
+const WalletKey = require('../lib/wallet/walletkey');
+
+describe('Node Sync', function() {
+  this.timeout(60000);
+
+  const mnemonic = [
+    'abandon', 'abandon', 'abandon', 'abandon',
+    'abandon', 'abandon', 'abandon', 'abandon',
+    'abandon', 'abandon', 'abandon', 'about'
+  ].join(' ');
+
+  const ports = {
+    p2p: 49331,
+    node: 49332,
+    wallet: 49333,
+    rs: 49334,
+    ns: 49335
+  };
+
+  const lookahead = 10;
+
+  let node, node2 = null;
+  let node2wallet = null;
+  let wdb, wdb2 = null;
+  let wallet, wallet2 = null;
+
+  before(async () => {
+    /**
+     * Setup initial nodes and wallets.
+     */
+
+    node = new FullNode({
+      memory: true,
+      apiKey: 'foo',
+      network: 'regtest',
+      workers: true,
+      workersSize: 2,
+      bip37: true,
+      plugins: [require('../lib/wallet/plugin')],
+      listen: true,
+      port: ports.p2p,
+      httpPort: ports.node,
+      rsPort: ports.rs,
+      nsPort: ports.ns,
+      env: {
+        'BCOIN_WALLET_HTTP_PORT': ports.wallet.toString()
+      }
+    });
+
+    await node.open();
+
+    node2 = new FullNode({
+      memory: true,
+      apiKey: 'foo',
+      network: 'regtest',
+      workers: true,
+      workersSize: 2,
+      port: ports.p2p + 5,
+      httpPort: ports.node + 5,
+      rsPort: ports.rs + 5,
+      nsPort: ports.ns + 5,
+      only: [`127.0.0.1:${ports.p2p}`]
+    });
+
+    await node2.open();
+
+    node2wallet = new WalletNode({
+      httpPort: ports.wallet + 5,
+      nodePort: ports.node + 5,
+      nodeApiKey: 'foo',
+      network: 'regtest'
+    });
+
+    await node2wallet.open();
+
+    /**
+     * Generate blocks and transactions.
+     */
+
+    await node.connect();
+
+    // Prepare the miner and wallet.
+    const {miner, chain} = node;
+    wdb = node.require('walletdb').wdb;
+    wallet = await wdb.create();
+    miner.addAddress(await wallet.receiveAddress());
+
+    // Mature the initial coins to use for the
+    // use in generating the test case.
+    for (let i = 0; i < 200; i++) {
```

Comment:
I think in hsd the coinbase maturity for regtest is much smaller so we can reduce this here.

https://github.com/handshake-org/hsd/blob/c85d9b4c743a9e1c9577d840e1bd20dee33473d3/lib/protocol/networks.js#L745